cinema/views.py

```diff
@@ -62,3 +98,53 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset.select_related("movie", "cinema_hall")
+        movies = self.request.query_params.get("movie")
+
+        if movies:
+            movies_ids = self._params_to_ints(movies)
+            queryset = self.queryset.filter(movie__id__in=movies_ids)
+
+        filter_date = self.request.query_params.get("date")
+
+        if filter_date:
+            queryset = queryset.filter(
+                show_time=datetime.strptime(filter_date, "%Y-%m-%d")
+            )
+
+        queryset = (queryset.annotate(
+            tickets_available=(F("cinema_hall__seats_in_row")
+                               * F("cinema_hall__rows")
+                               - Count("tickets"))))
+
+        return queryset
+
+
+class OrderPagination(PageNumberPagination):
+    page_size = 3
+    page_query_param = "page_size"
+    max_page_size = 100
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+    pagination_class = OrderPagination
+
+    def get_queryset(self):
+        queryset = self.queryset.filter(user=self.request.user)
+
+        if self.action == "list":
+            queryset = queryset.prefetch_related("tickets__movie_session")
```

Comment:
may be in this queryset will be better releted on the field "tickets__movie_session__cinema_hall"